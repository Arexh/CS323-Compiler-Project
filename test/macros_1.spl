#define PI 3.1415926535898
#define calc_area(radius); (radius) * (radius) * PI;

int main(){
    int r = 5;
    float area = calc_area(r);
    return 0;
}