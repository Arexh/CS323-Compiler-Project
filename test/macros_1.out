Program (1)
  ExtDefList (4)
    ExtDef (4)
      Specifier (4)
        TYPE: int
      FunDec (4)
        ID: main
        LP
        RP
      CompSt (4)
        LC
        DefList (5)
          Def (5)
            Specifier (5)
              TYPE: int
            DecList (5)
              Dec (5)
                VarDec (5)
                  ID: r
                ASSIGN
                Exp (5)
                  INT: 5
            SEMI
          DefList (6)
            Def (6)
              Specifier (6)
                TYPE: float
              DecList (6)
                Dec (6)
                  VarDec (6)
                    ID: area
                  ASSIGN
                  Exp (6)
                    Exp (6)
                      Exp (6)
                        ID: r
                      MUL
                      Exp (6)
                        ID: r
                    MUL
                    Exp (6)
                      FLOAT: 3.1415926535898
              SEMI
        StmtList (7)
          Stmt (7)
            RETURN
            Exp (7)
              INT: 0
            SEMI
        RC
